packages/EasyStandard/docs/ecs/sniffs.md

```diff
@@ -0,0 +1,410 @@
+    }
+    
+    public function testSomethingB()
+    {
+        $expectedResult = 4;
+        $actualResult = 2 + 2;
+        self::assertSame($expectedResult, $actualResult);
+    }
+}
+```
+```php
+// correct
+final class TestClass
+{
+    public function testSomethingA()
+    {
+        $expectedResult = 4;
+        $array = [
+            'key' => 'value',
+        ];
+
+        $actualResult = 2 + 2;
+
+        self::assertSame($expectedResult, $actualResult);
+        self::assertSame(['key' => 'value'], $array);
+    }
+    
+    public function testSomethingB()
+    {
+        $actualResult = 2 + 2;
+        
+        self::assertSame(4, $actualResult);
+    }
+
+    public function testSomethingC()
+    {
+        self::assertSame(4, 2 + 2);
+    }    
+
+    // Allow empty line in closure
+    public function testSomethingD()
+    {
+        $value1 = 2;
+        $value2 = 2;
+        $expectedClosure = static function () use ($value1, $value2): int {
+            $result = $value1 + $value2;
+
+            return $result + 0;
+        };
+
+        $actualResult = 2 + 2;
+
+        self::assertSame($expectedClosure(), $actualResult);
+    }
+
+    public function noTestMethod()
+    {
+        $expectedResult = 4;
+        $actualResult = 2 + 2;
+        self::assertSame($expectedResult, $actualResult);
+    }
+}
+```
+**Parameters**
+
+- `testMethodPrefix` - If a method name starts with this prefix, checks will be applied to it. Default value: `test`.
+- `testNamespace` - If a class namespace starts with this prefix, the class will be parsed. Default value: `App\Tests`.
+```php
+// correct
+namespace App\NoTestNamespace;
+
+final class TestClass
+{
+    public function testSomething()
+    {
+        $expectedResult = 4;
+
+        $actualResult = 2 + 2;
+
+        self::assertSame($expectedResult, $actualResult);
+
+        echo $actualResult;
+    }
+}
+```
+
+##### [\EonX\EasyStandard\Sniffs\ControlStructures\NoNotOperatorSniff][8]
+A strict comparison operator must be used instead of a NOT operator.
+```php
+// incorrect
+$a = (bool)\random_int(0, 1);
+if (!$a) {
+    // Do something.
+}
+````
+```php
+// correct
+$a = (bool)\random_int(0, 1);
+if ($a === false) {
+    // Do something.
+}
+````
+
+#### Exceptions
+
+##### [\EonX\EasyStandard\Sniffs\Exceptions\ThrowExceptionMessageSniff][9]
+Exception message must be either a variable or a translation message, started with one of valid prefixes.
+```php
+// incorrect
+throw new \Exception('Incorrect message');
+````
+```php
+// correct
+throw new NotFoundHttpException();
+// or
+$exception = new Exception('Some exception message');
+throw $exception;
+// or
+throw new InvalidArgumentException('exceptions.some_message');
+// or
+$message = 'Some exception message';
+throw new RuntimeException($message);
+````
+**Parameters**
+
+- `validPrefixes` - An array of prefixes to start with the message text line. Default value: `['exceptions.']`.
```

Comment:
Maybe try "An array of prefixes that are valid for starting the message text."

Reply:
Done.